CodeReview/Maru_code_forReview.py

```diff
@@ -0,0 +1,77 @@
+import numpy as np
+import pandas as pd
+import xarray as xr
+import requests
+
+from netCDF4 import Dataset
+
+
+def ferry_data_download(URL):
+    """Downloads the ADCP data from ferry
+    Inputs:
+    URL with the location of the ferry NetCDF file,
+    must end with .ncml
+    Outputs:
+    An object containing the link to the ferry data """
+    explanation = 'File exists'
+    file_downloaded = True
+    # Request if the thredds server is working, add .html to URL
+    req = requests.get(URL + '.html')
+    if req.status_code == 200:
+        """File exists and is good for download, so write file"""
+        print('File is ok')
+        explanation = 'Good URL File downloaded'
+        file_downloaded = True
+        ferry = xr.open_dataset(URL)
+    else:
+        print('File not found or unavailable')
+        explanation = ' File not found or unavailable'
+        file_downloaded = False
```

Comment:
If the file is not found, is ferry undefined? Not sure what the standard is here, but maybe return an empty ferry variable or something?